src/core/regimes.rkt

```diff
@@ -42,52 +42,37 @@
            (display ">" port))])
 
 (define (exprs-to-branch-on alts)
-  (define critexpr (critical-subexpression (*start-prog*)))
+  (define critexprs (all-critical-subexpressions (*start-prog*)))
   (define vars (program-variables (alt-program (car alts))))
 
-  (if critexpr
-      (remove-duplicates (cons critexpr vars))
+  (if critexprs
```

Comment:
Can this return `false` anymore? Wouldn't it just return the empty string? Also, it is ever possible for its list not to include the list of variables? It seems like it should be possible to remove the `if` statement and the `append`.